### DISCUSSION TITLE
🚀 Feature request: AVIF Image support

### DISCUSSION THREAD

**Discussion by SyntaxDreamer** [ExpressionEngine/ExpressionEngine#4506] [Feature Requests]
### Suggest an idea for ExpressionEngine

It seems AVIF is poised to become the new standard for images, WebP is great, but AVIF images are even smaller, saving a lot of disk space and bandwidth. WordPress, Joomla, and most CMS already support AVIF. And it is now supported in all major browsers including Safari, Chrome (and other Chromium browsers), Microsoft Edge and Firefox. Also on mobile, Android, Windows and recent MacOS as well Linux distros support it.

AVIF can also replace animated GIF's. It's not just a bit better than WebP, the difference is 5-10x in size without any quality reduction reason why designers and developers are demanding support for this extension and vendors have already delivered.

To enable AVIF uploads as images, all we need to edit in Expression Engine is the file:
`system/ee/ExpressionEngine/Config/mimes.php`

And add the extension, example, under the WebP mime type we add the new AVIF extension:

```
    'image/webp', // .webp
    'image/avif', // .avif
```

Now we can upload AVIF images with EE without issues. Except...

The Constraint or Crop image manipulations are not working because this probably needs some PHP work. I'm also not aware if other changes need to be done, but it is not new code, since the code that does image manipulation or supports WebP, PNG, etc, can be just replicated. With PHP we use the same libraries for manipulation, either GD or ImageMagick.

AVIF is supported since PHP 8.1 with the GD library that is basically installed and required by EE and most applications natively.

It's a must requirement to support AVIF out of the box in 2023 for any content management system. I did not find any open issue or request for AVIF which is surprising and this is the reason I'm opening this discussion for further feedback on the changes we need to apply to code and files.

**Comment by intoeetive** [comment]
Thank you for this request, I find this quite reasonable.

Please see the PR that should get it working https://github.com/ExpressionEngine/ExpressionEngine/pull/4837
